### ISSUE TITLE
'Operator 'NOT' requires two operands.'

### ISSUE THREAD

**Issue by nstdio** [yiisoft/yii2-mongodb#155]
### What steps will reproduce the problem?
```php
use yii\mongodb\Query;
// ...
$query
    ->andWhere(['not', $col, null])
    ->andWhere(['<>', $col, '']);
```
### What's expected?

The query will builded without error.

### What do you get instead?

`Fatal error: Uncaught exception 'yii\base\InvalidParamException' with message 'Operator 'NOT' requires two operands.' in vendor\yiisoft\yii2-mongodb\QueryBuilder.php on line 670`
```
1	0.0000	126968	{main}( )	...\index.php:0
2	0.0110	1186288	yii\base\Application->run( )	...\index.php:80
3	0.0110	1215912	yii\web\Application->handleRequest( )	...\Application.php:375
4	0.0120	1300592	yii\base\Module->runAction( )	...\Application.php:100
5	0.0140	1448736	yii\base\Controller->runAction( )	...\Module.php:454
6	0.0160	1633008	yii\base\InlineAction->runWithParams( )	...\Controller.php:154
7	0.0160	1634168	call_user_func_array:{D:\os\OpenServer\domains\yii-notymo\vendor\yiisoft\yii2\base\InlineAction.php:55} ( )	...\InlineAction.php:55
8	0.0160	1634320	nstdio\yii2notymo\controllers\SiteController->actionIndex( )	...\InlineAction.php:55
9	0.0220	2792024	nstdio\yii2notymo\PushNotification->send( )	...\SiteController.php:27
10	0.0220	2792192	nstdio\yii2notymo\provider\DataProvider->getTokens( )	...\PushNotification.php:63
11	0.0220	2795296	yii\mongodb\Query->all( )	...\DataProvider.php:69
12	0.0220	2795408	yii\mongodb\Query->buildCursor( )	...\Query.php:307
13	0.0240	2924520	yii\mongodb\Collection->find( )	...\Query.php:184
14	0.0250	3051464	yii\mongodb\Command->find( )	...\Collection.php:218
15	0.0260	3159272	yii\mongodb\QueryBuilder->buildCondition( )	...\Command.php:592
16	0.0260	3159560	yii\mongodb\QueryBuilder->buildAndCondition( )	...\QueryBuilder.php:619
17	0.0260	3159872	yii\mongodb\QueryBuilder->buildCondition( )	...\QueryBuilder.php:699
18	0.0260	3160136	yii\mongodb\QueryBuilder->buildAndCondition( )	...\QueryBuilder.php:619
19	0.0260	3160432	yii\mongodb\QueryBuilder->buildCondition( )	...\QueryBuilder.php:699
20	0.0260	3160696	yii\mongodb\QueryBuilder->buildAndCondition( )	...\QueryBuilder.php:619
21	0.0260	3160992	yii\mongodb\QueryBuilder->buildCondition( )	...\QueryBuilder.php:699
22	0.0260	3161256	yii\mongodb\QueryBuilder->buildAndCondition( )	...\QueryBuilder.php:619
23	0.0260	3161552	yii\mongodb\QueryBuilder->buildCondition( )	...\QueryBuilder.php:699
24	0.0260	3161848	yii\mongodb\QueryBuilder->buildNotCondition( )	...\QueryBuilder.php:619
```
### Additional info

| Q                      | A
| ---------------------- | ---
| Yii version            |  2.0.10-dev
| Yii MongoDB version    | 2.1.2
| MongoDB server version | 3.2 x64
| PHP version            | 5.6.19
| Operating system       | Windows 7 x64

**Comment by nstdio** [comment]
I've found some solution
```php
$query->andWhere(['not', $col, ['$eq' => null]])
```

But the problem still exists. What if I change `$query` implementation at runtime from `yii\mongodb\Query` to `yii\db\Query` and execute same query with code mentioned above.
I will send pull request soon.

**Comment by klimov-paul** [comment]
Resolved by commit ab8e4effea7ae72e20be8749fedfa0b79a2e42a8
